app/helpers/selections_helper.rb

```diff
@@ -1,2 +1,50 @@
+require 'json'
+require 'ostruct'
+
 module SelectionsHelper
+
+  def get_forecast(stations)
+
+    w_api = Wunderground.new("dda1e42fba07eb1c")
+    current_user.forecast = [];
+
+    @forecasts = []
+
+    # user forecast is an array of forecasts
+    current_user.selections.each do |selection|
```

Comment:
I think `each` could be `map` then you would not have to `@forecasts.push`. Just return the result of the `map`.

Reply:
Change `current_user.selections` to `selections` (from the method call).